scanner/src/main/kotlin/storages/ClearlyDefinedStorage.kt

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (C) 2020 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner.storages
+
+import java.io.IOException
+import java.lang.IllegalArgumentException
+import java.time.Instant
+
+import org.ossreviewtoolkit.clearlydefined.ClearlyDefinedService
+import org.ossreviewtoolkit.model.Failure
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.Provenance
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.Result
+import org.ossreviewtoolkit.model.ScanResult
+import org.ossreviewtoolkit.model.ScanResultContainer
+import org.ossreviewtoolkit.model.ScannerDetails
+import org.ossreviewtoolkit.model.Success
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsInfoCurationData
+import org.ossreviewtoolkit.model.config.ClearlyDefinedStorageConfiguration
+import org.ossreviewtoolkit.model.jsonMapper
+import org.ossreviewtoolkit.model.utils.toClearlyDefinedCoordinates
+import org.ossreviewtoolkit.model.utils.toClearlyDefinedSourceLocation
+import org.ossreviewtoolkit.scanner.ScanResultsStorage
+import org.ossreviewtoolkit.scanner.scanners.generateScannerDetails
+import org.ossreviewtoolkit.scanner.scanners.generateSummary
+import org.ossreviewtoolkit.utils.log
+
+import retrofit2.Call
+
+/** The name used by ClearlyDefined for the ScanCode tool. */
```

Comment:
So far we haven't used KDoc one-liner comments like these; I wasn't even aware they're supported. I somewhat like them for their compactness, but I'm wondering what's your rule to use one-liner comments, as e.g. in line 54 you didn't. Could you please elaborate?

Reply:
Do you maybe limit the use to properties?

Reply:
And I was not aware of a potential difference to Javadoc ;-)

I use this style for properties if the description is short enough.  For functions I am used to multi-line comments from my Java background. Here comments for methods normally span multiple lines as they contain tags for parameters and return values.